client/platform/desktop/backend/native/common.ts

```diff
@@ -827,10 +845,21 @@ async function _importTrackFile(
 
       const newPath = npath.join(projectDirAbsPath, npath.basename(newFileName));
 
-      await fs.copy(
-        trackFileAbsPath,
-        newPath,
-      );
+      if (jsonMeta.originalVideoFile && await nistSerializers.confirmNistFile(trackFileAbsPath)) {
+        const trackData = await nistSerializers.loadNistFile(trackFileAbsPath);
+        const trackStructure: MultiTrackRecord = {};
+        for (let i = 0; i < trackData.tracks.length; i += 1) {
+          const track = trackData.tracks[i];
+          trackStructure[track.trackId] = track;
+        }
+        const serialized = JSON.stringify(trackStructure);
+        await fs.writeFile(newPath, serialized);
+      } else {
+        await fs.copy(
+          trackFileAbsPath,
+          newPath,
+        );
```

Comment:
Looks like copy/paste from just a few lines up.  Why are `_importTrackFile` and `annotationImport` different functions?

Reply:
There may be a refactor that can combine them maybe with more arguments.  `_importTrackFile` is an initial import when loading a dataset so it is moving data around to place the `result_` in the proper location while extracting and updating jsonMeta for the import.

`annotationImport` is importing data into an already existing dataset so it acquires a lock and only places the file in the proper location with `result_` naming scheme so it will auto-load on refresh.

I'm just writing notes out below for myself when refactoring to combine the duplicate code

`_importTrackFile`:
- calculates and saves Metadata for loading
- Creates empty file if no file is specified
- Returns metadata for adding to localStorage

`annotationImport`:
- Determines if the specified file is valid track.json/NIST/csv
- acquires lock on existing track file
- deletes existing track file
- creates JSON from track file and moves/renames it so it can be autoloaded
- returns true to indicate successful import.

The easy thing to combine is the checking/generation of the track.json file that is used into something like `generateTrackJson` which will take a file path and either generate a track.json or return empty if it cna't find a valid format.,